sway-lib-std/src/bytes.sw

```diff
@@ -718,8 +723,10 @@ impl Bytes {
         // set capacity and length
         self.len = both_len;
 
-        // clear `other`
-        other.clear();
+        // clear `other`, if self == other then we do not want to clear self
+        if self.ptr() != other.ptr() {
```

Comment:
This is still UB. You can't use pointer equality to check unicity.

Reply:
If one cannot compare `self` and `other`, how do you suggest not clearing in this care or reverting in my original implementation? This is absolutely required to do.

Reply:
What I'm suggesting is that we do not clear `other` and not consume the appended values as Rust does. That way the algorithm doesn't have to branch on something that it can't branch on.

Where the pointer comes from doesn't matter, this would still result in different behavior is the compiler aliases things differently which is illegal.